Build failing when using  `cargo install --git`

Hello, thanks for the wonderful job.  For a couple of days I'm getting errors when trying to build the crate, here's the complete log:

```
$ cargo install --git https://github.com/rustsec/rustsec.git cargo-audit
    Updating git repository `https://github.com/rustsec/rustsec.git`
  Installing cargo-audit v0.17.4 (https://github.com/rustsec/rustsec.git#23884a0a)
    Updating crates.io index
   Compiling proc-macro2 v1.0.51
   Compiling quote v1.0.23
   Compiling unicode-ident v1.0.6
   Compiling syn v1.0.109
   Compiling autocfg v1.1.0
   Compiling libc v0.2.139
   Compiling serde_derive v1.0.152
   Compiling serde v1.0.152
   Compiling cfg-if v1.0.0
   Compiling pkg-config v0.3.26
   Compiling version_check v0.9.4
   Compiling memchr v2.5.0
   Compiling tinyvec_macros v0.1.1
   Compiling once_cell v1.17.1
   Compiling hashbrown v0.12.3
   Compiling tinyvec v1.6.0
   Compiling crossbeam-utils v0.8.15
   Compiling indexmap v1.9.2
   Compiling semver v1.0.16
   Compiling memoffset v0.8.0
   Compiling log v0.4.17
   Compiling unicode-bidi v0.3.10
   Compiling percent-encoding v2.2.0
   Compiling form_urlencoded v1.1.0
   Compiling crossbeam-epoch v0.9.14
   Compiling jobserver v0.1.26
   Compiling proc-macro-error-attr v1.0.4
   Compiling cc v1.0.79
   Compiling unicode-normalization v0.1.22
   Compiling serde_json v1.0.93
   Compiling regex-syntax v0.6.28
   Compiling scopeguard v1.1.0
   Compiling proc-macro-error v1.0.4
   Compiling adler v1.0.2
   Compiling bitflags v1.3.2
   Compiling rayon-core v1.10.2
   Compiling itoa v1.0.5
   Compiling ryu v1.0.12
   Compiling fixedbitset v0.4.2
   Compiling crossbeam-deque v0.8.3
   Compiling idna v0.3.0
   Compiling petgraph v0.6.3
   Compiling miniz_oxide v0.6.2
   Compiling crossbeam-channel v0.5.7
   Compiling num_cpus v1.15.0
   Compiling aho-corasick v0.7.20
   Compiling smartstring v1.0.1
   Compiling openssl-sys v0.9.80
   Compiling libz-sys v1.1.8
   Compiling libssh2-sys v0.2.23
   Compiling libgit2-sys v0.14.2+1.5.1
   Compiling backtrace v0.3.67
   Compiling tracing-core v0.1.30
   Compiling gimli v0.27.2
   Compiling eyre v0.6.8
   Compiling lazy_static v1.4.0
   Compiling regex v1.7.1
   Compiling regex-automata v0.1.10
   Compiling object v0.30.3
   Compiling indenter v0.3.3
   Compiling unicode-xid v0.2.4
   Compiling static_assertions v1.1.0
   Compiling openssl-probe v0.1.5
   Compiling thiserror v1.0.38
   Compiling winnow v0.3.3
   Compiling either v1.8.1
   Compiling platforms v3.1.0 (/home/edu4rdshl/.cargo/git/checkouts/rustsec-37bc61f9328ed4f7/23884a0/platforms)
   Compiling overload v0.1.1
   Compiling pin-project-lite v0.2.9
   Compiling rustc-demangle v0.1.21
   Compiling heck v0.4.1
   Compiling addr2line v0.19.0
   Compiling os_str_bytes v6.4.1
   Compiling clap_lex v0.2.4
   Compiling nu-ansi-term v0.46.0
   Compiling rayon v1.6.1
   Compiling synstructure v0.12.6
   Compiling matchers v0.1.0
   Compiling tracing-log v0.1.3
   Compiling sharded-slab v0.1.4
   Compiling atty v0.2.14
   Compiling thread_local v1.1.7
   Compiling zeroize v1.5.7
   Compiling rustc-hash v1.1.0
   Compiling tracing-attributes v0.1.23
   Compiling clap_derive v3.2.18
   Compiling thiserror-impl v1.0.38
   Compiling humantime v2.1.0
   Compiling binfarce v0.2.1
   Compiling ident_case v1.0.1
   Compiling fs-err v2.9.0
   Compiling strsim v0.10.0
   Compiling textwrap v0.16.0
   Compiling topological-sort v0.2.2
   Compiling termcolor v1.2.0
   Compiling home v0.5.4
   Compiling tracing v0.1.37
   Compiling owo-colors v3.5.0
   Compiling smallvec v1.10.0
   Compiling auditable-extract v0.3.2
   Compiling abscissa_derive v0.6.0
   Compiling tracing-subscriber v0.3.16
   Compiling wait-timeout v0.2.0
   Compiling canonical-path v2.0.2
   Compiling arc-swap v1.6.0
   Compiling clap v3.2.23
   Compiling color-eyre v0.6.2
   Compiling url v2.3.1
   Compiling toml v0.5.11
   Compiling toml_datetime v0.6.1
   Compiling serde_spanned v0.6.1
   Compiling hex v0.4.3
   Compiling toml_edit v0.19.4
   Compiling cvss v2.0.0 (/home/edu4rdshl/.cargo/git/checkouts/rustsec-37bc61f9328ed4f7/23884a0/cvss)
   Compiling humantime-serde v1.1.1
   Compiling secrecy v0.8.0
   Compiling quitters v0.1.0 (/home/edu4rdshl/.cargo/git/checkouts/rustsec-37bc61f9328ed4f7/23884a0/quitters)
   Compiling git2 v0.15.0
   Compiling cargo-lock v8.0.3
   Compiling abscissa_core v0.6.0
   Compiling auditable-serde v0.5.2
   Compiling toml v0.7.2
   Compiling auditable-info v0.6.2
   Compiling crates-index v0.18.11
   Compiling rustsec v0.26.4 (/home/edu4rdshl/.cargo/git/checkouts/rustsec-37bc61f9328ed4f7/23884a0/rustsec)
error[E0282]: type annotations needed
  --> rustsec/src/repository/git/commit.rs:63:27
   |
63 |                 Some(data.as_ref().into()),
   |                           ^^^^^^   ---- type must be known at this point
   |
help: try using a fully qualified path to specify the expected types
   |
63 |                 Some(<[u8] as AsRef<T>>::as_ref(data).into()),
   |                      +++++++++++++++++++++++++++    ~

For more information about this error, try `rustc --explain E0282`.
error: could not compile `rustsec` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-audit v0.17.4 (https://github.com/rustsec/rustsec.git#23884a0a)`, intermediate artifacts can be found at `/tmp/cargo-install9LWo63`
```

I have tried with both the stable tool chain and the nightly tool chain with the same result. Doing a manual `git clone` and then `cargo build --bin cargo-audit` works... so weird.

--- a/rustsec/src/repository/git/commit.rs
+++ b/rustsec/src/repository/git/commit.rs
@@ -4,7 +4,7 @@ use crate::{
     error::{Error, ErrorKind},
     repository::{git::Repository, signature::Signature},
 };
-use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use std::{time::{Duration, SystemTime, UNIX_EPOCH}, ops::Deref};
 
 /// Number of days after which the repo will be considered stale
 /// (90 days)
@@ -60,7 +60,7 @@ impl Commit {
         let (signature, signed_data) = match repo.repo.extract_signature(&oid, None) {
             Ok((ref sig, ref data)) => (
                 Some(Signature::from_bytes(sig)?),
-                Some(data.as_ref().into()),
+                Some(data.deref().into()),
             ),
             _ => (None, None),
         };

--- a/rustsec/src/repository/git/commit.rs
+++ b/rustsec/src/repository/git/commit.rs
@@ -4,7 +4,10 @@ use crate::{
     error::{Error, ErrorKind},
     repository::{git::Repository, signature::Signature},
 };
-use std::{time::{Duration, SystemTime, UNIX_EPOCH}, ops::Deref};
+use std::{
+    ops::Deref,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
 
 /// Number of days after which the repo will be considered stale
 /// (90 days)
@@ -58,10 +61,9 @@ impl Commit {
             .to_owned();
 
         let (signature, signed_data) = match repo.repo.extract_signature(&oid, None) {
-            Ok((ref sig, ref data)) => (
-                Some(Signature::from_bytes(sig)?),
-                Some(data.deref().into()),
-            ),
+            Ok((ref sig, ref data)) => {
+                (Some(Signature::from_bytes(sig)?), Some(data.deref().into()))
+            }
             _ => (None, None),
         };
 

